src/YarpUtilities/include/BipedalLocomotionControllers/YarpUtilities/Helper.h

```diff
@@ -0,0 +1,149 @@
+/**
+ * @file Helper.h
+ * @authors Giulio Romualdi
+ * @copyright 2019 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTROLLERS_YARP_UTILITIES_HELPER_H
+#define BIPEDAL_LOCOMOTION_CONTROLLERS_YARP_UTILITIES_HELPER_H
+
+// std
+#include <deque>
+#include <vector>
+
+// YARP
+#include <yarp/os/BufferedPort.h>
+#include <yarp/os/Property.h>
+#include <yarp/os/Value.h>
+#include <yarp/sig/Vector.h>
+
+#include <iDynTree/Core/VectorDynSize.h>
+
+// clang-format off
+
+#define YARP_UTILITES_GET_ELEMENT_TYPE(type)                            \
+    ((std::is_same<type, double>::value) ? "double" :                   \
+    ((std::is_same<type, int>::value) ? "int" :                         \
+    ((std::is_same<type, std::string>::value) ? "string" :              \
+    ((std::is_same<type, bool>::value) ? "bool" :                       \
+     "undefined" ))))
+
+#define YARP_UTILITES_CHECK_ELEMENT_SUPPORT(type)                       \
+    ((std::is_same<type, double>::value) ? true :                       \
+    ((std::is_same<type, int>::value) ? true :                          \
+    ((std::is_same<type, std::string>::value) ? true :                  \
+    ((std::is_same<type, bool>::value) ? true :                         \
+     false ))))
+
+#define YARP_UTILITES_GET_CHECKER_NAME(type)                                                  \
+    ((std::is_same<type, int>::value) ? &yarp::os::Value::isInt :                             \
+    ((std::is_same<type, double>::value) ? &yarp::os::Value::isDouble :                       \
+    ((std::is_same<type, std::string>::value) ? &yarp::os::Value::isString :                  \
+    ((std::is_same<type, bool>::value) ? &yarp::os::Value::isBool :                           \
+     &yarp::os::Value::isDouble ))))
```

Comment:
Maybe you can move them to the ``tpp`` file.

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/5/commits/6cfdf5d1dbcbbb91c418cc94bf890731fc9df9bf